aptos-move/block-executor/src/view.rs

```diff
@@ -1067,28 +1146,232 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TResourceGr
         &self,
         group_key: &Self::GroupKey,
         resource_tag: &Self::ResourceTag,
-        _maybe_layout: Option<&Self::Layout>,
+        maybe_layout: Option<&Self::Layout>,
     ) -> anyhow::Result<Option<Bytes>> {
-        // TODO[agg_v2]: support layouts.
+        // println!(
+        //     "get_resource_from_group {:?} {:?} {:?}",
+        //     group_key, resource_tag, maybe_layout
+        // );
+        let maybe_layout = maybe_layout.map(|layout| Arc::new(layout.clone()));
+
```

Comment:
I think there is an issue here how we treat Unset and patching together. Also code here seems longer than necessary, let me try something here.

we need more tests as sell for these functions in view.rs